### ISSUE TITLE
Some kind of "unsubscribe" from Q&A threads

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#1610]
We're starting to get threads that are long and active enough that people want to unsubscribe, like this one:

https://publiclab.org/questions/warren/09-08-2017/how-do-you-merge-gps-logger-data-into-photographs#answer-385-comment

This level of email notification is actually a fairly new feature and we haven't implemented an unsubscribe yet. GitHub has the "Not Watching/Watching/Ignoring" resolution but also the "following this thread" one... needs some thinking through.

The current system looks for /all/ thread participants, using this method for new comments:

https://github.com/publiclab/plots2/blob/master/app/models/comment.rb#L106

And this one for new answers:

https://github.com/publiclab/plots2/blob/master/app/models/answer.rb#L27

If we were to offer an unsubscribe, we might need to change this system to actually add people as "likers" or "followers" when they comment or answer, so we could delete their "following" status even though they participated in the thread, and we don't delete their comment or answer itself.

This could be complex and we may want to start by looking at how the "subscribe to thread" mechanism works, using `node_selection` rather than directly counting comments.

**Comment by jywarren** [comment]
@jlev thanks!

**Comment by jywarren** [comment]
@Gauravano is solving this in https://github.com/publiclab/plots2/pull/3119, i believe!

**Comment by Gauravano** [comment]
You got it right @jywarren

**Comment by Gauravano** [comment]
Actually, #3119 is a small part of this unsubscribe system, we want to build. Currently, a user can unsubscribe from the the
1. Like notifications (user being author)
2. Comment notifications (user being author)
3. Indirect comment notification (user being commenter)

@jywarren we need to add more points to this, what do you think?

**Comment by jywarren** [comment]
hmm, answer notifications? what other kinds are there?

**Comment by Gauravano** [comment]
yes answer notifications and unsubscribing from a node like @shapironick mentioned here https://github.com/publiclab/plots2/issues/2910#issuecomment-407800414 . What do you think?

**Comment by jywarren** [comment]
agreed, so this is 2 additional types of unsubscribe?

Would Nick's suggestion be like, `commented on a note but no longer want to be updated` -- and how would we store that? Could we... store a "silence this thread" somewhere?

**Comment by jywarren** [comment]
nodeselection could work, would we set a true/false column added to
nodeselection? and we can hide the button under the Like menu maybe?

And add a link in comment emails themselves?

**Comment by Gauravano** [comment]
The button can be added at many places, I guess we can come to that later. I will first see all places where NodeSelection is used, so that we don't mess up with any previous feature/function. Thanks!

**Comment by jywarren** [comment]
Hmm, I think we ended up using "liking" to mean both liking and following.
Perhaps we should make pressing the button enable both a like and a follow,
but then if you want you can "unselect" following so you don't take away
the star but you don't get updates anymore?

**Comment by Gauravano** [comment]
ok, so we are using `following`, for saving status. Ok, so I will raise a PR for migration as we need to change the following of all existing nodeselection rows to true and also we need to set default to true. What do you think @jywarren ? Thanks!

**Comment by Gauravano** [comment]
@jywarren so can I go for the above strategy?

**Comment by jywarren** [comment]
that sounds great!

**Comment by gauravano** [comment]
Checklist for the Node Un-subscription: 
- [ ] Make a migration for changing the default to true for new NodeSelection entries
- [ ] Migration for changing `following` column in NodeSelection to true for present rows in DB
- [ ] A controller method for handling subscribing and unsubscribing to node
- [ ] before sending email, a check which would ensure that email should not be sent to the user if he has un-subscribed to that node
- [ ] test for Emails

@jywarren I am following this checklist for this issue. Please add/change if I am forgetting something. Thanks!

**Comment by jywarren** [comment]
This looks good! Thanks!
